libraries/chain/dbs_registration_committee.cpp

```diff
@@ -0,0 +1,141 @@
+#include <scorum/chain/dbs_registration_committee.hpp>
+#include <scorum/chain/database.hpp>
+
+#include <scorum/chain/dbs_account.hpp>
+
+namespace scorum {
+namespace chain {
+
+dbs_registration_committee::dbs_registration_committee(database& db)
+    : _base_type(db)
+{
+}
+
+dbs_registration_committee::registration_committee_member_refs_type dbs_registration_committee::get_committee() const
+{
+    registration_committee_member_refs_type ret;
+
+    const auto& idx = db_impl().get_index<registration_committee_member_index>().indicies();
+    for (auto it = idx.cbegin(); it != idx.cend(); ++it)
+    {
+        ret.push_back(std::cref(*it));
+    }
+
+    return ret;
+}
+
+const registration_committee_member_object&
+dbs_registration_committee::get_member(const account_name_type& account) const
+{
+    try
+    {
+        return db_impl().get<registration_committee_member_object, by_account_name>(account);
+    }
+    FC_CAPTURE_AND_RETHROW((account))
+}
+
+dbs_registration_committee::registration_committee_member_refs_type
+dbs_registration_committee::create_committee(const genesis_state_type& genesis_state)
+{
+    FC_ASSERT(!genesis_state.registration_committee.empty(), "Registration committee must have at least one member.");
+    FC_ASSERT(SCORUM_REGISTRATION_LIMIT_COUNT_COMMITTEE_MEMBERS > 0, "Invalid ${1} value.",
+              ("1", SCORUM_REGISTRATION_LIMIT_COUNT_COMMITTEE_MEMBERS));
+
+    // check existence here to allow unit tests check input data even if object exists in DB
+    FC_ASSERT(get_committee().empty(), "Can't create more than one committee.");
+
+    const dbs_account& account_service = db().obtain_service<dbs_account>();
+
+    // create unique accout list form genesis unordered data
+    using sorted_type = std::map<account_name_type, std::reference_wrapper<const account_object>>;
+    sorted_type items;
+    for (const auto& genesis_item : genesis_state.registration_committee)
+    {
+        const account_object& accout = account_service.get_account(genesis_item);
+
+        items.insert(sorted_type::value_type(accout.name, std::cref(accout)));
+    }
+
+    // add members
+    for (const auto& item : items)
+    {
+        if (_get_member_count() > SCORUM_REGISTRATION_LIMIT_COUNT_COMMITTEE_MEMBERS)
+        {
+            wlog("Too many committee members in genesis state. More than ${1} are ignored.",
+                 ("1", SCORUM_REGISTRATION_LIMIT_COUNT_COMMITTEE_MEMBERS));
+            break;
+        }
+        const account_object& accout = item.second;
+        _add_member(accout);
+    }
+
+    if (!_get_member_count())
+    {
+        FC_ASSERT(false, "Can't initialize at least one member.");
+    }
+
+    return get_committee();
+}
+
+const registration_committee_member_object&
+dbs_registration_committee::add_member(const account_name_type& account_name)
+{
+    FC_ASSERT(!get_committee().empty(), "No committee to add memeber.");
```

Comment:
Why we can't add new member if committee has no members yet?

Reply:
But who will vote?